litestar/stores/file.py

```diff
@@ -88,7 +88,7 @@ async def set(self, key: str, value: str | bytes, expires_in: int | timedelta |
             ``None``
         """
 
-        await self.path.mkdir(exist_ok=True)
+        await self.path.mkdir(exist_ok=True, parents=True)
```

Comment:
if this change is being introduced, would it be better to make it configurable by a bool in `FileStore.__init__`?

Reply:
Yes. This absolutely needs to be configurable and set to `False` by default. Otherwise it's not only a breaking change, but can also lead to nasty side effects

Reply:
Why creating 1 folder is fine, but creating 2 has to be configurable. The "allow_create_folders" flag makes sense to me. "allow_create_more_than_1_folder" makes no sense to me.
Or I misunderstand the flag's purpose?

Reply:
Please follow the guidance given here.  It is definitely a breaking change and has security implications.